File: 69-sqrtx/69-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0 or x == 1:
            return x
        left = 0
        right = x
        while left <= right:
            mid = left + (right-left) // 2
            square = mid * mid
            if square == x:
                return mid
            if square < x:
                left =mid + 1
            else:
                right = mid -1
        return left -1

File: 69-sqrtx/69-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        long long low = 1 , high = x ,  ans = -1;
        if (x==0 || x == 1)
            return x;
        while (low <= high){
            long long mid = (low + high)/2;
            long long sq = mid * mid;
            if (sq == x)
                return mid;
            else if(sq > x)
                high = mid - 1;
            else
            {
                low = mid + 1;
                ans = mid;
            }
        }
        return ans;
    }
};